File: beginner/pagina02/1047/1047.php
<?php 

    list($h1, $m1, $h2, $m2) = explode(" ", readline());

    $h1 = (int) $h1; 
    $m1 = (int) $m1; 
    $h2 = (int) $h2; 
    $m2 = (int) $m2; 

    $hr = 0;
    $min = 0;

    if($h2 < $h1 || ($h2 == $h1 && $m2 < $m1)){
        $h2 += 24;
    }

    if($m2 < $m1){
        $m2 += 60;
        $h2 -= 1;
    }

    $hr = $h2 - $h1;
    $min = $m2 - $m1;

    if($hr == 0 && $min == 0){
        $hr = 24;
        $min = 0;
    }

    echo "O JOGO DUROU $hr HORA(S) E $min MINUTO(S)\n";
?>

File: beginner/pagina02/1047/1047.rs
use std::io;

fn main() {
    let mut intput = String::new();

    io::stdin().read_line(&mut input).unwrap();

    let data: Vec<u32> = input.trim().split_whitespace()
                     .map(|x| x.parse::<u32>().unwrap())
                     .collect();

    let (mut h1, mut m1, mut h2, mut m2) = (data[0], datat[1], data[2], data[3]);

    if h2 < h1 ||  (h2 == h1 && m2 <= m1) {
        h2 += 24;
    }

    if m2 < m1 {
        m2 += 60;
        h2 -= 1;
    }

    let s1 = (h1 * 3600) + (m1 * 60);
    let s2 = (h2 * 3600) + (m2 * 60);

    let tempo = s2 - s1;
    let hr = tempo / 3600;
    let min = (tempo % 3600) / 60;

    println!("O JOGO DUROU {} HORA(S) E {} MINUTO(S)", hr, min);
}